lib/elixir/lib/calendar/iso.ex

```diff
@@ -71,6 +91,23 @@ defmodule Calendar.ISO do
       ]
     end
 
+  defguardp is_year(year) when year in -9999..9999
+  defguardp is_year_BCE(year) when year in -9999..0
+  defguardp is_year_CE(year) when year in 1..9999
+  defguardp is_month(month) when month in 1..12
+  defguardp is_day(day) when day in 1..31
+  defguardp is_hour(hour) when hour in 0..23
+  defguardp is_minute(minute) when minute in 0..59
+  defguardp is_second(second) when second in 0..59
+
+  defguardp is_microsecond(microsecond, precision)
+            when microsecond in 0..999_999 and precision in 0..6
```

Comment:
I really like your solution here of passing two arguments and leaving the destructuring up to the caller. :+1: